### PR TITLE
Add Multi-dof joint state in storage and retrieval services

### PR DESCRIPTION
**Pull request by Abishalini**

### CODE REVIEW

File: moveit_studio_msgs/moveit_studio_agent_msgs/msg/RobotJointState.msg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Contains joint state information for robots with all types of Joints (REVOLUTE, PRISMATICS, PLANAR, FLOATING)
+sensor_msgs/JointState joint_state
```
Comment (marioprats) [inline moveit_studio_msgs/moveit_studio_agent_msgs/msg/RobotJointState.msg:2]:
```suggestion
# Contains the joint positions of single-dof joints (i.e. REVOLUTE, PRISMATIC).
sensor_msgs/JointState joint_state

```

File: moveit_studio_msgs/moveit_studio_agent_msgs/msg/RobotJointState.msg [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+sensor_msgs/MultiDOFJointState multi_dof_joint_state
```
Comment (marioprats) [inline moveit_studio_msgs/moveit_studio_agent_msgs/msg/RobotJointState.msg:3]:
```suggestion
# Contains the joint positions of multi-dof joints (i.e. PLANAR, FLOATING).
sensor_msgs/MultiDOFJointState multi_dof_joint_state
```

**Review by marioprats — APPROVED** [review]
LGTM. I added some additional doc as suggestions.
Did you consider reusing the `moveit_msgs/RobotState` message instead of a new one? That one already contains `joint_state` and `multi_dof_joint_state`, although it also contains some other fields that we don't need. So I think it makes sense to have a custom message with just the things we need.

**Comment by Abishalini** [comment]
Good point about resuing `RobotState`. 
For now, I will keep this new `RobotJointState` msg and refactor when we need to pass the attached collision objects information.

**Pull request merged**